src/types/Course/Preference.ts

```diff
@@ -35,38 +73,216 @@ export const PreferenceQuery = extendType({
           },
         });
 
-        const courses = prefs.map(async ({ courseID, rank }) => {
-          const courseData = await (prisma as PrismaClient).course.findUnique({
-            where: {
-              id: courseID,
-            },
-          });
+        const courses = await Promise.all(
+          prefs.map(async ({ courseID, rank }) => {
+            const courseData = await (prisma as PrismaClient).course.findUnique({
+              where: {
+                id: courseID,
+              },
+            });
 
-          if (!courseData) {
+            if (!courseData) {
+              return {
+                id: {
+                  subject: '',
+                  code: '',
+                  term: '' as Term,
+                  year: 0,
+                },
+                preference: 0,
+              };
+            }
             return {
               id: {
-                subject: '',
-                code: '',
-                term: '' as Term,
-                year: 0,
+                subject: courseData.subject,
+                code: courseData.code,
+                term: courseData.term,
+                year: courseData.year,
               },
-              preference: 0,
+              preference: rank ?? 0,
             };
-          }
-          return {
-            id: {
-              subject: courseData.subject,
-              code: courseData.code,
-              term: courseData.term,
-              year: courseData.year,
-            },
-            preference: rank ?? 0,
-          };
-        });
+          })
+        );
         return {
           courses,
         };
       },
     });
   },
 });
+
+export const AllPreferencesQuery = extendType({
+  type: 'Query',
+  definition(t) {
+    t.list.nonNull.field('coursePreferences', {
+      type: CoursePreference,
+      description: 'Get all courses preferences',
+      resolve: async (_, __, { prisma }) => {
+        const prefs = await (prisma as PrismaClient).preference.findMany();
```

Comment:
Here you could do something like
```typescript
const courses = await (prisma as PrismaClient).course.findMany({
  include: {
     preferences: true
  }
});
```
Then the `preferences` will be accessible like `courses[0].preferences` for example. Then you can map that data to match what you're returning but that might simplify this resolver a bit,